---
title: "Final exam"
output:
  pdf_document:
    toc: true
  html_notebook:
    toc: true
    toc_float: true
    number_sections: false
editor_options:
  markdown:
    wrap: 72
---

```{r echo=FALSE,message=FALSE, warning=FALSE}
library(ggplot2)
library(emmeans)
library(lme4)
library(lmerTest)
library(multcomp)
library(PLS205)
```

# Instructions

1.  Use the "Visual" markdown version of this document. Click "Visual"
    in the top-left corner of this document before starting.
2.  All necessary packages are loaded in the above code block. Check if
    they all load by clicking the Run arrow
3.  Fill in your answers (R code and text) in each block provided.
    -   Only questions that include an empty code block need R code
    -   All questions need a text answer. Please delete the text "ENTER
        YOUR RESPONSE HERE" and replace with your answer.
4.  When you are done, click the drop-down menu next to "**Preview**"
    and select "**Knit to PDF**". Check the pdf to make sure all R code
    and any associated output is provided.
5.  Upload both the pdf to Gradescope.
6.  This exam is to be done independently. Please do not consult with
    others in the class. You can use any external resource available to
    you. But we will grade based on the material I presented in class.
7.  If you have clarification questions:
    1.  Check Piazza to see if the question has been addressed already
    2.  If not, email the professor or TAs. We will post answers to
        Piazza so that everyone has access to the clarifications. Please
        do not post directly so as not to give away answers to others.
8.  Read every question **completely** and **carefully**! Below the bold
    title-text the notes and instructions are very important to
    answering correctly.

------------------------------------------------------------------------

# Question 1 - part 1

![](Almond_covers.jpeg)

A yield trial was performed at an almond orchard to evaluate different
management treatments for the ground between trees. **3 different ground
covers (Cover)** were evaluated. The orchard was divided into **5 strips
of trees (Strip)**, and then each Strip was divided into **3 sections
(Section)**. Each section of each strip was assigned one of the three
ground cover treatments. The **total yield of almonds (yield, kg)**
harvested from trees in the middle of each section was measured at the
end of the season.

```{r}
almonds = read.csv('almonds_1_field.csv')
str(almonds)
```

## 1.1 Create a design table for this experiment [8 points]

Give a justification for each EU and Block that you specify.

**NOTE:** Right click on a row to insert new rows if needed. You can
also make the table in Excel and paste your table directly here.

| Structure | Variable    | Type        | \# levels | Block | EU      |
|-----------|-------------|-------------|-----------|-------|---------|
| Treatment | Cover       | Categorical | 3         | Strip | Section |
| Design    | Strip       | Categorical | 5         |       |         |
|           | Cover:Strip | Categorical | 15        |       |         |
|           | Section     | Categorical | 15        |       |         |
| Response  | yield       | Numeric     | 15        |       |         |

> My block is `Strip`, since it has one section for each of my three
> treatment levels, and they are effectively replicates. The EU is
> `Section`, since each one only receives one level of treatment (as
> opposed to the three treatments each `Strip` receives).

## 1.2 Write an appropriate linear model for the analysis [4 points]

Be sure to fix any variables in the data table.

```{r}
almonds$Section = as.factor(almonds$Section)
almodel = lm(yield ~ Cover + Strip, data = almonds)
```

> Needed to switch `Section` to something not-numeric. Other than that,
> used `lm`, since there isn't actually replication of each individual
> `Cover:Strip`.

## 1.3 Can you conclude that any of the ground covers affect yield? Which covers appear best? Worst? [6 points]

Use alpha = 0.05. Show the R output used to answer the question and
reference the specific values you use in your text.

```{r}
anova(almodel)

almeans = emmeans(almodel, spec = 'Cover')
covereffects = contrast(almeans, method = 'pairwise')
coversumm = summary(covereffects, infer = T, alpha = 0.05)
print(almeans)
print(coversumm)
```

> Yes, judging by an anova, we can confidently say that at least one of
> the covers affects yield (p-value 9.23e-06). The best performing
> covers appear to be either A or B, and the worst is C, though A and B
> are not really distinguishable from each other.

# Question 1 - part 2

Given these promising results, you decide to expand the experiment by
adding two additional orchards. You replicate the same design over those
two additional orchards and add them to your data table.

```{r}
almonds_3_Field = read.csv('almonds_3_field.csv',stringsAsFactors = TRUE)
str(almonds_3_Field)
```

## 1.4 Create a new design table for this larger experiment [8 points]

Give a justification for each EU and Block that you specify.

| Structure | Variable          | Type        | \# levels | Block        | EU      |
|-----------|-------------------|-------------|-----------|--------------|---------|
| Treatment | Cover             | Categorical | 3         | Strip, Field | Section |
| Design    | Strip             | Categorical | 5         |              |         |
|           | Field             | Categorical | 3         |              |         |
|           | Strip:Field       | Categorical | 15        |              |         |
|           | Cover:Strip:Field | Categorical | 45        |              |         |
|           | Section           | Categorical | 45        |              |         |
| Response  | yield             | Numeric     | 45        |              |         |

> Cover is blocked both in `Strip` and `Field`, both receiving the full
> set of Treatments. The EU is still `Section`, since it is still the
> only part of the design that receives a single treatment.

## 1.5 Write an appropriate linear model for the analysis [4 points]

Be sure to fix any variables in the data table.

```{r}
almonds_3_Field$Section = as.factor(almonds_3_Field$Section)
threemod = lmer(yield ~ Field + Cover + (1|Strip), data = almonds_3_Field)

```

> Using `lmer`, since `Strip` should be a random variable.

## 1.6 Estimate the effects of Cover again, and compare the results of this larger experiment to the first experiment. [6 points]

Have your conclusions changed? You've done 3x the work. Have you gained
precision in your estimates relative to the first (smaller experiment)?
How has the *interpretation* of your treatment effects changed with the
new experiment? Show the effects tables used to answer the question and
reference the specific values you use in your text.

```{r}
anova(threemod, ddf = 'K')
threemeans = emmeans(threemod, spec='Cover')
threeeffects = contrast(threemeans, method = 'pairwise')
threesumm = summary(threeeffects, infer = T, level=1-0.05/3)
threesumm$p.value = pmin(1, threesumm$p.value * 3)
print(threemeans)
print(threesumm)
no_blocks = lm(yield ~ Cover + Strip, data = almonds_3_Field)
anova(no_blocks)
```

> My conclusions have not changed, but the context has. The SE of our
> interaction effects is 4.81 as opposed to 2.15 in the previous
> experiment, which leads to bigger CIs, and less confidence in the
> average effect. Truthfully, it's not making much sense to add the two
> extra fields unless you were explicitly interested to see if they
> behaved differently (which doesn't seem to be the intention).
> Theoretically, adding the blocks can indicate if `Field` is
> interacting with `Cover` and changing treatment effects, but even with
> pretending as if `Field`s don't exist and we didn't block, the p-value
> of `Cover` in the anova is nearly the same as with the block structure
> (0.0005535 vs 0.0005977). Without adding `Field` as a treatment, the
> replication hasn't improved our precision nor the strength of our
> interpretation.
>
> However, were we to consider `Field` as a treatment (in addition to a
> block), we could look make statements about whether it appears that
> our `Cover` treatment effect is affected by `Field`. Again, when
> pretending `Field` isn't a block, there isn't a dramatic shift in the
> F-value or p-value of `Cover` in the anova table. While the blocks
> ultimately would not have done anything, we would be more confident in
> generalizing our results to other, similar fields.

# Question 2

![](Sorghum.jpg)

An experiment was done to test whether the gene PHOSPHORUS-STARVATION
TOLERANCE1 (SbPSTOL1) affects Sorghum's ability to take up enough
phosphate to maintain root growth under limited phosphate conditions.
**Four independent mutants (`MutA`, `MutB`, `MutC`, and `MutD`)** that
each independently break the gene SbPSTOL1 in different ways were
developed using CRISPR/Cas9 base editing in the r**eference sorghum
genotype `Tx430`**. The experiment used hydroponics, meaning that the
plants were suspended in a liquid medium where specific concentrations
of nutrients could be accurately controlled. Specifically, **10 tubs**
(each 20in x 12 in) were filled with Magnavaca modified nutrient
solution with either **2.5 uM (low-P) or 250 uM (high-P) phosphate**
added as KH2PO4. Plants were germinated on filter paper and then
transferred to plugs in the lid of each bin and grown for three weeks.
**Each tub housed 15 plants, 7 of the reference genotype `Tx430`, and
two each of the four mutants**. The solution was continuously aerated
and changed every 5 days. At the end of the experiment each plant was
removed, the roots excised, dried, and individually weighed and total
dry root mass (g) was recorded for each plant.

```{r}
sorghum = read.csv('sorghum_root.csv')
str(sorghum)
```

## 2.1 Prepare a Design Table for this experiment [8 points]

The focal treatment is `Geno`. Give a justification for each EU and
Block that you specify.

**NOTE:** Right click on a row to insert new rows if needed. You can
also make the table in Excel and paste your table directly here.

| Structure | Variable   | Type        | N. levels | Block           | EU    |
|-----------|------------|-------------|-----------|-----------------|-------|
| Focal     | Geno       | Categorical | 5         | Phos, Tub       | Plant |
| Moderator | Phos       | Categorical | 2         |                 | Tub   |
| Combo     | Geno:Phos  | Categorical | 10        | Tub(Incomplete) | Plant |
| Design    | Tub        | Categorical | 10        |                 |       |
|           | Plant:Phos | Categorical | 30        |                 |       |
|           | Plant      | Categorical | 150       |                 |       |
| Response  | RootMass   | Numeric     | 150       |                 |       |

> Okay, so with a focal treatment of `Geno`, our EU is Plant, since
> that's the part of the design that receives only one level of that
> treatment. `Geno` is blocked both in `Tub`, since each one contains
> every level of the treatment, as does each level of `Phos` (which is
> almost uninteresting. Of course the focal treatment is blocked in the
> moderator). Our EU for `Phos` is `Tub`, for a similar reason as
> before. Each `Tub` receives only one level of `Phos`. There is no
> block for `Phos`. Our combo variable, `Geno:Phos`, has an EU of plant,
> once again, since that is the only part of the design that receives
> only one level of the treatment. It is blocked incompletely by `Tub`.
> While each `Tub` does not receive the full suite of treatments, pairs
> of hi-P and low-P treated `Tub`s represent the full range.
>
> `Phos:Tub` and `Tub` are aliased, and since `Tub` is an EU, I'm going
> to keep it over the other.

## 2.2 Write an appropriate linear model for the analysis [4 points]

Be sure to fix any variables in the data table. Be careful about nested
terms.

```{r}
sorghum$Tub = as.factor(sorghum$Tub)
sorghum$Plant = as.factor(sorghum$Plant)
sorgmod = lmer(RootMass ~ (1|Tub) + (1|Plant:Phos) + Geno + Phos + Geno:Phos, data = sorghum)

```

> Using both `Tub` and `Plant:Phos` as my random terms in the model.

## 2.3 Evaluate whether any of the mutants alter root growth in either Phosphorous treatment relative to Tx430. Use $\alpha = 0.05$. Do not use an ANOVA. [6 points]

Show the effects table used to answer the question and reference the
specific values you use in your text.

```{r}
sormeans = emmeans(sorgmod, spec = 'Geno', by = 'Phos')
sorcomp = contrast(sormeans, method = 'trt.vs.ctrl', ref = 'Tx430', name = 'Gen_effect')
sorsumm = summary(sorcomp, infer = T, level = 1-0.05/2, as.df = TRUE)
sorsumm$p.value = pmin(1, sorsumm$p.value * 2)
print(sorsumm)
```

> Yes, we can see that there are at least a few mutants that later root
> growth relative to Tx430 (when using a Bonferroni multiple comparison
> correction at 2, since that is how many levels our moderator has),
> MutA and MutC at alpha = 0.05. The p-value for both at low-P is below
> 0.0001, and at the hi-P MutC also appears to alter root growth with a
> p-value == 0.0001.

## 2.4 Evaluate which mutant effects on root growth changed in low Phosphorous vs High Phosphorous conditions. Use $\alpha = 0.05$. [6 points]

Show the effects table used to answer the question and reference the
specific values you use in your text.

```{r}
#now comparison
regroupsor = update(sorcomp, by = 'Gen_effect')
sorinter = contrast(regroupsor, method = 'pairwise')
sorintersumm = summary(sorinter, infer = T, level = 1-0.05/4, as.df = TRUE)
sorintersumm$p.value = pmin(1, sorintersumm$p.value * 4)
print(sorintersumm)
```

> Again using a Bonferroni correction (This time at 4, since we have 4
> comparisons across each specific effect), we can see that the only
> mutant effects on root growth that really changed was MutA, with a
> p.value of below 0.001.

## 2.5 Discuss what the researchers learned about the role of SbPSTOL1 on root growth. [5 points]

Address the following:

1.  You should have found one mutant with a strong effect in both
    conditions, and one mutant with an effect that changed considerably
    between conditions. Which mutant is more interesting for follow-up
    work? Why?
2.  Given that each mutation can alter the genome in other ways in
    addition to "breaking" a gene, and that each mutation breaks the
    gene in different ways, is it reasonable to conclude from this work
    that breaking the gene SpPSTOL1 will affect phosphorous responses in
    all Sorghum varieties? Why or why not? What would you recommend to
    the researchers to gain further evidence that this gene is actually
    involved?

Note: A figure might help you explain your answer to #1.

> 1.  The researchers found both MutA and MutC. However, MutC's effect
>     didn't seem to be affected by the levels of Phos, whereas MutA
>     was. This makes MutA a little more interesting than MutC, since it
>     is a **conditional knockout**. A knockout can answer questions
>     about what happens when a gene breaks completely, but conditional
>     knockouts have more uses beyond that. They can be used to model
>     effects of scarcity, as a method of establishing a control vs
>     treatment relationship without having to create two separate
>     transgenic plant lines, and more. Furthermore, it is likely that
>     effect on the structure of the protein for knockout of MutA is a
>     little more subtle than the full knockout of MutC. Even if the
>     *opposite* is true, this could provide some interesting insights
>     into the protein structure of SbPSTOL1 and its relationship with
>     function. (i.e., which domain in the protein has the most effect
>     on phosphate uptake, etc).
>
> <!-- -->
>
> 2.  No. Off target effects are off target effects, and can change
>     depending on what strain you are doing your experiment in. There
>     is no real replication or manipulation of the, essentially,
>     `Mutation:PlantLine` combination. These effects, therefore, can
>     only really be stated with confidence that they occur in the Tx430
>     line. Each individual strain of sorghum will have its own unique
>     genetic landscape (it's why they're different strains to begin
>     with); in order to make the statement that knocking out/down
>     SbPSTOL1 will affect phosphorous responses in all Sorghum
>     varieties, they would need to repeat this experiment in different
>     strains, then compare the specific effects of a SbPSTOL1 k/o on
>     each strain.

## 2.6 Speculate on why the researcher allocated more plants to the Tx430 genotype in each tub than to each of the mutants. Was this a reasonable choice to make? [5 points]

> I have two guesses here. The first is that creating transgenic plant
> lines is not a simple task. It is time-consuming and potentially
> expensive and error prone. The second is that more control treated
> plants means your estimation of the baseline interaction between
> `Phos` and `Geno` is more confident; i.e. you have a better idea of
> the baseline effect that switching phosphorous concentrations does.
> This in turn means we can make more confident predictions about the
> effect of different Mutants with less total plants assigned to those
> treatments. I think this was reasonable, assuming they were limited in
> space, time, and funds.
